File: frontend/src/api/page.ts
import { http } from './http';

export interface PagePayload {
  title: string;
  urlPath: string;
  content: string;
  status?: 'DRAFT' | 'PUBLISHED';
}

export const pageApi = {
  list() {
    return http.get<PagePayload[]>('/pages');
  },
  get(id: number) {
    return http.get<PagePayload>(`/pages/${id}`);
  },
  create(payload: PagePayload) {
    return http.post<PagePayload>('/pages', payload);
  },
  update(id: number, payload: PagePayload) {
    return http.put<PagePayload>(`/pages/${id}`, payload);
  },
  remove(id: number) {
    return http.delete<void>(`/pages/${id}`);
  },
};

File: backend/src/main/java/com/ainativecms/entity/Page.java
package com.ainativecms.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Table;

@Entity
@Table(name = "pages")
public class Page extends BaseEntity {

    @Column(nullable = false)
    private String title;

    @Column(nullable = false, unique = true)
    private String urlPath;

    @Column(columnDefinition = "TEXT")
    private String content;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Status status = Status.DRAFT;

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getUrlPath() {
        return urlPath;
    }

    public void setUrlPath(String urlPath) {
        this.urlPath = urlPath;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public Status getStatus() {
        return status;
    }

    public void setStatus(Status status) {
        this.status = status;
    }

    public enum Status {
        DRAFT,
        PUBLISHED
    }
}
